### PR TITLE
Finish Google-OAuth2.0 Implementation

### PR DESCRIPTION
**Pull request by Kenttrann2302**
Pull request description:

+ Finish implementing google-oauth2.0 using flask and pyopenssl for https certificate, and create a route for logging out the user.

+ In general, the logic should work as user can sign in with their google account and the response with status 201 successfully. However, still in demand to write some tests to let user get into studyhub dashboard.

+ This pull request is done and can be viewed now before merging in master.

### CODE REVIEW

File: database/users_models.py [hunk 1]
```diff
@@ -236,7 +245,9 @@ def __repr__(self):
 for i in range(12):
     list_study_env.append(study_env["Study_env_preferences"][f"{i+1}"])
     list_study_time.append(
-        study_time["Study_time_preferences"][f"{i+1}"]["Time"] + " " + study_time["Study_time_preferences"][f"{i+1}"]["Frame"]
+        study_time["Study_time_preferences"][f"{i+1}"]["Time"]
+        + " "
+        + study_time["Study_time_preferences"][f"{i+1}"]["Frame"]
```
Comment (j-ngo828) [inline database/users_models.py:250]: let's use f-string here:

```suggestion
        f"{study_time["Study_time_preferences"][f"{i+1}"]["Time"]} {study_time["Study_time_preferences"][f"{i+1}"]["Frame"]}"
```

File: requirements.txt
Comment (j-ngo828) [inline requirements.txt:1]: Add the following to `requirements.old.txt`:

```
Flask-Login==0.4.1
oauthlib==3.0.1
pyOpenSSL==19.0.0
```

Then try creating a new virtual environment, and run `pip install -r requirements.old.txt`. Then retest things again to see if it works.

If it works, great, then we can setup Docker. Else, well let's hope it won't get to this else.

File: login.py
Comment (j-ngo828) [inline login.py:1]: 🔥

**Review by j-ngo828 — CHANGES_REQUESTED** [review]
Just some very minor change on the code, and updating the `requirements.old.txt`

I recommend creating a Postman collection for testing this OAuth registration flow, as well as the existing user authentication flow in addition to CRUD operations on profile and study preferences. It'd make it easier to demo the API to employers in the upcoming weeks.

After you follow the steps related to new virtual environment with `requirements.old.txt` below, and everything works. Pick up the issue that mentions using Docker for dev environment. I will give instruction for that in the issue

File: requirements.txt
Comment (Kenttrann2302) [reply requirements.txt:1]: I don't know why when I created a new virtual environment and tried to download the requirements.old.txt, there are some dependencies that are not included in the file such as pyobjc-... were installed as well?

Comment (Kenttrann2302) [reply requirements.txt:1]: @j-ngo828

Comment (Kenttrann2302) [reply requirements.txt:1]: I did deactivate my previous virtual environment and create a new virtual environment, after that I run this command: pip install -r requirements.old.txt and I saw the pyobjc,... (dependencies that are not there being installed as well).

Comment (j-ngo828) [reply requirements.txt:1]: I'm going to assume you created a new virtual environment named `new-env`.

Are `pyobjc` and other unwanted dependencies in `new-env/lib/{python-version}/site-packages`? {python-version} is the python version used to create the virtual environment, if it's python 3.11 then {python-version} is `python3.11`.

If not then ignore the log output of `pip install`. If it's in there then I have 0 idea why and how to fix this. A solution is using a different package manager like Pipenv and convert the `requirements.old.txt` file accordingly

Comment (Kenttrann2302) [reply requirements.txt:1]: Okay thank you, I'll try my best today to find out the root of the problem

Comment (j-ngo828) [reply requirements.txt:1]: Don't worry about it now, this is outside the scope of this issue.

It's in the scope of the Docker container issue. Just add the dependencies to `requirements.old.txt` and make some other changes I suggested above and push it so I can merge and approve.

Worry about the problem in #19

Comment (Kenttrann2302) [reply requirements.txt:1]: Okay, I just pushed the final commit for this PR. We will figure out a way to resolve the dependencies issue and Docker container setup later after I finished the final PR for the final REST API.

**Review by j-ngo828 — APPROVED** [review]
LGTM

**Pull request merged**